R/call_utils.R

```diff
@@ -49,108 +21,96 @@ call_check_parse_varname <- function(varname) {
 
-  if (length(choices) == 1) {
-    call("==", varname, choices)
+  if (is.numeric(choices)) {
+    if (length(choices) == 1) {
+      sprintf("%s == %s", varname, sprintf("%s", choices))
+    } else {
+      sprintf("%s %%in%% c(%s)", varname, toString(sprintf("%s", choices)))
+    }
   } else {
-    c_call <- do.call(
-      "call",
-      append(list("c"), choices)
-    )
-    # c_call needed because it needs to be vector call
-    # instead of vector. SummarizedExperiment.subset
-    # handles only vector calls
-    call("%in%", varname, c_call)
+    if (length(choices) == 1) {
+      sprintf("%s == %s", varname, sprintf("\"%s\"", choices))
+    } else {
+      sprintf("%s %%in%% c(%s)", varname, toString(sprintf("\"%s\"", choices)))
+    }
   }
 }
 
 #' `numeric` range condition call
 #'
 #' Compose `numeric` range condition call from inputs
 #'
-#' @param varname (`name` or `character(1)`)\cr
+#' @param varname (`character(1)`)\cr
 #'   name of the variable
 #'
 #' @param range (`numeric(2)`)\cr
 #'   range of the variable
 #'
 #' @return call
 #' @examples
-#' teal.slice:::call_condition_range("AGE", range = c(1, 2))
-#' teal.slice:::call_condition_range(as.name("AGE"), range = c(-1.2, 2.1))
-#' teal.slice:::call_condition_range(
-#'   teal.slice:::call_extract_list("ADSL", "AGE"),
+#' \dontrun{
+#' call_condition_range("AGE", range = c(1, 2))
+#' call_condition_range(
+#'   call_extract_list("ADSL", "AGE"),
 #'   range = c(-1.2, 2.1)
 #' )
-#' @return a `call`
+#' }
+#' @return a character string
 #' @keywords internal
 call_condition_range <- function(varname, range) {
+  checkmate::assert_string(varname)
   checkmate::assert_numeric(range, len = 2, sorted = TRUE)
-  varname <- call_check_parse_varname(varname)
-  call(
-    "&",
-    call(">=", varname, range[1]),
-    call("<=", varname, range[2])
-  )
+
+  sprintf("%s >= %.4f & %s <= %.4f", varname, range[1], varname, range[2])
 }
 
 #' `logical` variable condition call
 #'
 #' Compose `logical` variable condition call from inputs
 #'
-#' @param varname (`name` or `character(1)`)\cr
+#' @param varname (`character(1)`)\cr
 #'   name of the variable
 #'
 #' @param choice (`logical(1)`)\cr
 #'   chosen value
 #'
 #' @return call
 #' @examples
-#' teal.slice:::call_condition_logical("event", choice = TRUE)
-#' teal.slice:::call_condition_logical("event", choice = FALSE)
-#' @return a `call`
+#' \dontrun{
+#' call_condition_logical("event", choice = TRUE)
+#' call_condition_logical("event", choice = FALSE)
+#' }
+#' @return character string
 #' @keywords internal
 call_condition_logical <- function(varname, choice) {
+  checkmate::assert_string(varname)
   checkmate::assert_flag(choice)
-  varname <- call_check_parse_varname(varname)
 
-  if (choice) {
-    varname
-  } else if (!choice) {
-    call("!", varname)
-  } else {
-    stop(
-      "Unknown filter state", toString(choice),
-      " for logical var ", as.character(varname)
-    )
-  }
+  sprintf("%s == %s", varname, choice)
```

Comment:
I personally prefer `!varname` instead of `varname != logical`. The most robust is `isTRUE(varname)` which wouldn't need `!is.na(varname)` if any value in the vector is missing

Reply:
`isTRUE` only returns TRUE for a logical of length 1.

I don't mind `!logical` but this string construction is neater. If not for multiple types in `call_condition_choice`, all of these `sprintf` calls could be moved to the respective `$get_call` methods and these functions could be removed.

Reply:
Rolled back to `!logical`.